### PR TITLE
feat(pci): add snapshot tasks

### PR DESCRIPTION
**Pull request by marie-j**
### Description of the Change

* Add onboarding for new feature
* Display tasks
* Allow to create/delete tasks 

WEBSITE-1680

# Needs

- [x] Review from Content team for the texts

### CODE REVIEW

File: client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.controller.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+      this.instanceId = this.$stateParams.instanceId;
+      this.instancesAreLoading = true;
+
+      this.model = {
+        action: null,
+        instance: null,
+        schedule: null,
+      };
+
+      this.getRegionsWithWorkflowService();
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.controller.js]:
```js
return this.getRegionsWithWorkflowService()
    .then(() => this. getCloudProjectEligibleInstances())
    .catch(() => {
        this.CloudMessage.error(this.$translate.instant('cpc_tasks_resources_error'));
          this.instancesAreLoading = false;
        });
    })
    .finally(() => {
        this.instancesAreLoading = false;
    });

```

File: client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.controller.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    isInstanceSelected() {
+      return !_.isNull(this.model.instance);
+    }
+
+    isScheduleSelected() {
+      return !_.isNull(this.model.schedule);
+    }
+
+    createTask() {
+      if (this.model.action.id === 'snapshot') {
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.controller.js]: You will add a `if` for each kind of task ? You could call the function according to a convention instead
```js
this[`generate${kindOfTask}TaskParams`]();
```

File: client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.controller.js [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        this.generateSnapshotTaskParams();
+        this.createSnapshotTask();
+      }
+    }
+
+    selectSchedule() {
+      this.isCustomSchedule = this.model.schedule === 'custom';
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.controller.js]: You creation could be divided into more than one entities (one for choosing the kind of task, one for selecting a schedule, one for each step afterwards)

File: client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.routes.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+angular.module('managerApp').config(($stateProvider) => {
+  $stateProvider.state('iaas.pci-project.compute.task.create', {
+    url: '/create',
+    views: {
+      'cloudProjectCompute@iaas.pci-project.compute': {
+        templateUrl: 'app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.html',
+        controller: 'CloudProjectComputeTasksCreateCtrl',
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/tasks/create/cloud-project-compute-tasks-create.routes.js]: Could have created a Component

File: client/app/cloud/project/compute/tasks/onboarding/cloud-project-compute-tasks-onboarding.html [hunk 5]
```diff
@@ -0,0 +1,22 @@
+<section class="cui-page__content">
+    <div class="text-center" data-ng-if="$ctrl.loading">
+        <oui-spinner data-size="m"></oui-spinner>
+    </div>
+    <div class="oui-slideshow oui-slideshow_onboarding oui-box" data-ng-if="!$ctrl.loading">
+        <div class="text-center my-auto">
+            <h2 class="oui-header_2" data-translate="cpc_tasks_onboarding_new"></h2>
+            <div class="oui-slideshow-panel__text">
+                <span class="oui-icon oui-icon-clock-wait" aria-hidden="true"></span>
```
Comment (FredericEspiau) [inline client/app/cloud/project/compute/tasks/onboarding/cloud-project-compute-tasks-onboarding.html]: No text substitute for this icon

File: client/components/validator/crontabExpression.factory.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+angular.module('managerApp').factory('CrontabExpression', () => {
```
Comment (FredericEspiau) [inline client/components/validator/crontabExpression.factory.js]: Isn't there a library to do this ?

File: client/app/cloud/project/compute/tasks/onboarding/cloud-project-compute-tasks-onboarding.html (hunk 5 above)
Comment (marie-j) [reply client/app/cloud/project/compute/tasks/onboarding/cloud-project-compute-tasks-onboarding.html]: It is just for cosmetic purpose

File: client/components/validator/crontabExpression.factory.js (hunk 6 above)
Comment (marie-j) [reply client/components/validator/crontabExpression.factory.js]: I wish there was

File: client/app/cloud/project/compute/tasks/onboarding/cloud-project-compute-tasks-onboarding.html (hunk 5 above)
Comment (FredericEspiau) [reply client/app/cloud/project/compute/tasks/onboarding/cloud-project-compute-tasks-onboarding.html]: Really ? It doesn't convey the information "please wait" ?

Comment (marie-j) [reply client/app/cloud/project/compute/tasks/onboarding/cloud-project-compute-tasks-onboarding.html]: I was asked to add a `metering icon`, among the ui-kit icons this one made sense given the context

**Comment by jleveugle** [comment]
Hello @marie-j,
Your pull request is now closed due to inactivity.
Feel free to re-open it if you have any update.
Anna Bot for OVH UX <3

**Pull request closed without merging**